generators/base-application/support/prepare-relationship.mjs

```diff
@@ -78,11 +99,13 @@ export default function prepareRelationship(entityWithConfig, relationship, gene
         return otherSideRelationship.relationshipName === relationship.otherEntityRelationshipName;
       });
       if (!otherRelationship) {
-        // TODO throw error at v8.
+        // TODO throw error at v8. - Check for ManyToMany with User entity: this produces a warning at the moment and maybe shouldn't then throw here but behave differently
         generator.logger.warn(
           `Error at '${entityName}' definitions: 'otherEntityRelationshipName' is set with value '${relationship.otherEntityRelationshipName}' at relationship '${relationship.relationshipName}' but no back-reference was found at '${otherEntityName}'`
         );
       } else if (
+        // renaming a relationship could cause trouble here - old relationship needs to be removed
+        !ignoreMissingRequiredRelationship &&
```

Comment:
Why is this?

Reply:
in case of a changed name in a existing relationship, we come to this if here. But since the new relationship name does not match anymore to the one on the other side (for some reasons). So, we would end up throwing the exception here if we weren't to ignore it also in this if statement and wouldn't go on removing the old relationship. I'll add another test that will only succeed if this addition is here ;-)

Reply:
I've added a test for that.